src/Security/CMSSecurity.php

```diff
@@ -163,28 +170,11 @@ public function GetLoginForms()
     public static function enabled()
     {
         // Disable shortcut
-        if (!static::config()->reauth_enabled) {
+        if (!static::config()->get('reauth_enabled')) {
             return false;
         }
 
-        // Count all cms-supported methods
-        $authenticators = Authenticator::get_authenticators();
-        foreach ($authenticators as $authenticator) {
-            // Supported if at least one authenticator is supported
-            if ($authenticator::supports_cms()) {
-                return true;
-            }
-        }
-        return false;
-    }
-
-    public function LoginForm()
-    {
-        $authenticator = $this->getAuthenticator();
-        if ($authenticator && $authenticator::supports_cms()) {
-            return $authenticator::get_cms_login_form($this);
-        }
-        user_error('Passed invalid authentication method', E_USER_ERROR);
+        return count(Security::singleton()->getApplicableAuthenticators(Authenticator::CMS_LOGIN));
```

Comment:
Lots of duplication here with Security::getAuthenticators(). Can't you just call that?